File: leetcode/design-parking-system.java
//https://leetcode.com/problems/design-parking-system/

class ParkingSystem {
    private int[] spotsRemaining;
    public ParkingSystem(int big, int medium, int small) {
        spotsRemaining = new int[]{big, medium, small};
    }
    
    public boolean addCar(int carType) {
        if(spotsRemaining[carType-1] >0){
            spotsRemaining[carType-1]--;
            return true;
        }
        return false;
    }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem obj = new ParkingSystem(big, medium, small);
 * boolean param_1 = obj.addCar(carType);
 */

File: leetcode/design_parking_system.py
# https://leetcode.com/problems/design-parking-system


class ParkingSystem:
    spots: list[int] = []

    def __init__(self, big: int, medium: int, small: int):
        self.spots = [big, medium, small]

    def addCar(self, carType: int) -> bool:
        self.spots[carType - 1] -= 1
        return self.spots[carType - 1] >= 0
